test/batch_test.rb

```diff
@@ -9,10 +9,13 @@ def test_batch
   end
 
   def test_nested_batch
-    GraphQL::Batch.batch do
-      assert_raises(GraphQL::Batch::NestedError) do
-        GraphQL::Batch.batch {}
+    product = GraphQL::Batch.batch do
+      promise = RecordLoader.for(Product).load(1)
+      GraphQL::Batch.batch do
+        RecordLoader.for(Product).load(1)
       end
+      promise
```

Comment:
You should also test a load here to make sure the executor is still setup.  Again, this could just be asserting that `RecordLoader.for(Product).load(1)` returns the same promise.

Reply:
Done.